packages/pluggableWidgets/rich-text-web/src/utils/useEditorSettings.ts

```diff
@@ -0,0 +1,118 @@
+
+const contentFilter: ConfigReducer = (config, props) => {
+    if (props.advancedContentFilter === "custom") {
+        config.allowedContent = props.allowedContent;
+        config.disallowedContent = props.disallowedContent;
+    }
+
+    return config;
+};
+
+contentFilter.getDeps = props => [props.advancedContentFilter];
+
+const plugins: ConfigReducer = (config, props) => {
+    if (props.codeHighlight) {
+        addPlugin("codesnippet", config);
+    }
+
+    return config;
+};
+
+plugins.getDeps = props => [props.codeHighlight];
+
+const toolbar: ConfigReducer = (config, props) => {
+    const { advancedConfig, toolbarConfig, preset } = props;
+    if (preset !== "custom") {
+        return Object.assign(config, getPreset(preset));
+    } else {
+        const groupKeys = Object.keys(props).filter((key: string) => (key.includes("Group") ? key : null));
+        let groupItems: ToolbarItems[] | string[];
+        if (toolbarConfig === "basic") {
+            groupItems = groupKeys
+                .filter((groupName: GroupType) => props[groupName])
+                .map((groupName: GroupType) =>
+                    groupName.includes("separator") ? "/" : groupName.replace("Group", "").toLowerCase()
+                );
+        } else {
+            groupItems = defineAdvancedGroups(advancedConfig);
+        }
+
+        return Object.assign(config, createCustomToolbar(groupItems, toolbarConfig === "basic"));
+    }
+};
+
+toolbar.getDeps = props => [props.advancedConfig, props.toolbarConfig, props.preset];
+
+const configReducers = [mapProps, editorWidth, toolbar, plugins, contentFilter];
+
+function createConfig(props: RichTextContainerProps): CKEditorConfig {
+    const initConfig: CKEditorConfig = {
+        autoGrow_minHeight: 300,
+        toolbarCanCollapse: true,
+        autoGrow_onStartup: true
+    };
+
+    return configReducers.reduce((config, fn) => fn(config, props), initConfig);
+}
+
+type EditorSettingsKey = string;
+
+type UseEditorSettingsResult = [RichTextSettings, EditorSettingsKey];
+
+/**
+ * This hook is intended to calculate CKEditor config.
+ * As CKEditor don't allow dynamic configuration, along with
+ * config we need calculate key. Key will be used to remount editor
+ * every time config is changed.
+ */
+export function useEditorSettings(props: RichTextContainerProps): UseEditorSettingsResult {
+    const getDependencies = (): any[] => {
+        const selectors = [() => [props.editorType, props.sanitizeContent], ...configReducers.map(fn => fn.getDeps)];
+
+        return selectors.map(fn => fn(props)).flat();
+    };
+    return useMemo(
+        () => [
+            {
+                type: props.editorType,
+                config: createConfig(props),
+                sanitizeContent: props.sanitizeContent
+            },
+            Date.now().toString()
+        ],
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        getDependencies()
```

Comment:
I got lost in the context, can you please explain why do you use getDependencies instead of just spreading props here?

Reply:
In short, Idea was to create set of "config reducers" -- functions that map RichText props to CKEditor configuration. As there many of this "config reducers" and it's hard to control all their dependencies in one place (in our case this will be dpes array in `useMemo`) I added special `getDeps` prop for each reducer. `getDeps` returns array of deps for current reducer. So, in turn, to make sure we always have "fresh" conifg, we combining this `getDeps` function to create one big dependencies array. This is why `getDependencies` function was created.